http/search.go

```diff
@@ -80,8 +83,22 @@ func (h *SearchHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		}
 	}
 
+	// check for timeout and create context
+	var ctx context.Context
+	timeoutStr := req.FormValue("timeout")
+	if timeoutStr == "" {
+		ctx = context.Background()
+	} else {
+		timeout, err := strconv.Atoi(timeoutStr)
```

Comment:
Instead of this, you can use the ParseDuration function from golang's time package.

```

// ParseDuration parses a duration string.
// A duration string is a possibly signed sequence of
// decimal numbers, each with optional fraction and a unit suffix,
// such as "300ms", "-1.5h" or "2h45m".
// Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
func ParseDuration(s string) (Duration, error) {
```

Reply:
Will use